src/main/scala/chiseltest/internal/BackendInterface.scala

```diff
@@ -59,15 +63,50 @@ trait BackendInterface { this: ThreadedBackend =>
 
     doPeek(signal, new Throwable)
     val dataName = dataNames(signal)
-    val a = simulatorInterface.peek(dataName) match {
+    val interfaceResult = simulatorInterface.peek(dataName) match {
       case Some(peekValue) =>
         logger.debug(s"${resolveName(signal)} -> $peekValue")
         peekValue
       case None =>
         logger.debug(s"${resolveName(signal)} is eliminated by firrtl, default 0.")
         BigInt(0)
     }
-    a
+
+    simulatorInterface match {
```

Comment:
This class should really be independent of particular interfaces, each backend should implement interface functions in a way that returns the data in a standardized format instead of matching on the particular backend in a function that should be agnostic of backends.

Reply:
Will fix.

Reply:
No more `BackendInterface` now.
Added API in `SimulatorInterface` to resolve this.
```resolvePeek(data: Data, interfaceResult: BigInt): BigInt```